File: src/extras-ksyntaxhighlighting/libabstracthighlighter.cpp
#define WORKAROUND_INNER_CLASS_DEFINITION_KSyntaxHighlighting__AbstractHighlighter
#include <KSyntaxHighlighting/Definition>
#include <KSyntaxHighlighting/FoldingRegion>
#include <KSyntaxHighlighting/Format>
#include <KSyntaxHighlighting/Theme>
#include <abstracthighlighter.h>
#include "libabstracthighlighter.h"
#include "libabstracthighlighter.hxx"

KSyntaxHighlighting__Definition* KSyntaxHighlighting__AbstractHighlighter_Definition(const KSyntaxHighlighting__AbstractHighlighter* self) {
    return new KSyntaxHighlighting::Definition(self->definition());
}

void KSyntaxHighlighting__AbstractHighlighter_SetDefinition(KSyntaxHighlighting__AbstractHighlighter* self, const KSyntaxHighlighting__Definition* def) {
    self->setDefinition(*def);
}

KSyntaxHighlighting__Theme* KSyntaxHighlighting__AbstractHighlighter_Theme(const KSyntaxHighlighting__AbstractHighlighter* self) {
    return new KSyntaxHighlighting::Theme(self->theme());
}

void KSyntaxHighlighting__AbstractHighlighter_SetTheme(KSyntaxHighlighting__AbstractHighlighter* self, const KSyntaxHighlighting__Theme* theme) {
    self->setTheme(*theme);
}

void KSyntaxHighlighting__AbstractHighlighter_Delete(KSyntaxHighlighting__AbstractHighlighter* self) {
    delete self;
}

File: include_original/extras-ksyntaxhighlighting/libabstracthighlighter.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");

/// https://api.kde.org/ksyntaxhighlighting-abstracthighlighter.html
pub const ksyntaxhighlighting__abstracthighlighter = struct {
    /// [Qt documentation](https://api.kde.org/ksyntaxhighlighting-abstracthighlighter.html#definition)
    ///
    /// ``` self: QtC.KSyntaxHighlighting__AbstractHighlighter ```
    pub fn Definition(self: ?*anyopaque) QtC.KSyntaxHighlighting__Definition {
        return qtc.KSyntaxHighlighting__AbstractHighlighter_Definition(@ptrCast(self));
    }

    /// [Qt documentation](https://api.kde.org/ksyntaxhighlighting-abstracthighlighter.html#setDefinition)
    ///
    /// ``` self: QtC.KSyntaxHighlighting__AbstractHighlighter, def: QtC.KSyntaxHighlighting__Definition ```
    pub fn SetDefinition(self: ?*anyopaque, def: ?*anyopaque) void {
        qtc.KSyntaxHighlighting__AbstractHighlighter_SetDefinition(@ptrCast(self), @ptrCast(def));
    }

    /// [Qt documentation](https://api.kde.org/ksyntaxhighlighting-abstracthighlighter.html#theme)
    ///
    /// ``` self: QtC.KSyntaxHighlighting__AbstractHighlighter ```
    pub fn Theme(self: ?*anyopaque) QtC.KSyntaxHighlighting__Theme {
        return qtc.KSyntaxHighlighting__AbstractHighlighter_Theme(@ptrCast(self));
    }

    /// [Qt documentation](https://api.kde.org/ksyntaxhighlighting-abstracthighlighter.html#setTheme)
    ///
    /// ``` self: QtC.KSyntaxHighlighting__AbstractHighlighter, theme: QtC.KSyntaxHighlighting__Theme ```
    pub fn SetTheme(self: ?*anyopaque, theme: ?*anyopaque) void {
        qtc.KSyntaxHighlighting__AbstractHighlighter_SetTheme(@ptrCast(self), @ptrCast(theme));
    }

    /// Delete this object from C++ memory.
    ///
    /// ``` self: QtC.KSyntaxHighlighting__AbstractHighlighter ```
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.KSyntaxHighlighting__AbstractHighlighter_Delete(@ptrCast(self));
    }
};
